modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ParentContentSynchronizer.java

```diff
@@ -246,14 +258,38 @@ else if ( actualTargetContext != null )
             {
                 actualSourceContext = sourceContexts.get( (NodePath) actualTargetContext.getAttribute( CONTENT_ROOT_PATH_ATTRIBUTE ) );
             }
+            else
+            {
+                return null;
+            }
+
+            if ( targetContent != null && ( targetContent.getOriginProject() == null ||
+                !actualSourceContext.getRepositoryId().equals( targetContent.getOriginProject().getRepoId() ) ) )
+            {
+                return null; // do not sync contents without origin parent or from other source
```

Comment:
This is a tricky part and must be covered with test

Reply:
removed it, it's unreal behavior outside of multi-inheritance POC for now